stream-api/src/main/kotlin/fund/cyber/markets/api/common/IncomingMessagesHandler.kt

```diff
@@ -37,9 +39,20 @@ class IncomingMessagesHandler(
                 }
             }
             is TradeChannelSubscriptionCommand -> {
-                tradesBroadcastersIndex.broadcastersFor(command.pairs, command.exchanges)
-                        ?.forEach { broadcaster -> broadcaster.registerChannel(wsChannel)
+                val broadcasters = tradesBroadcastersIndex.broadcastersFor(command.pairs, command.exchanges)
+                val result : LinkedList<Trade> = LinkedList()
+                var attemptCounter = 0
+                while(result.size < 10 && attemptCounter<30 ) {
+                    val randomTrade = broadcasters.toMutableList()[Int.rand(0, broadcasters.size)]
+                            .getRandomTradeFromBroadcaster()
+                    if(randomTrade!=null){
+                        val unique = result.none { it.tradeId == randomTrade.tradeId }
```

Comment:
we can skip this check, just add to result
low probability here + tradeid is not unique for whole trades, it is unique only for <TokensPair, Exchange> pair